### ISSUE TITLE
Can't create signatures 

### ISSUE THREAD

**Issue by sdgamboa** [waldronlab/bugsigdbr#42]
I don't think I could create signatures using `bugsigdbr::getSignatures`. Fresh installation. Code below:

``` r
library(bugsigdbr)
## BiocManager::install('bugsigdbr', force = TRUE)
BiocManager::valid()
#> 'getOption("repos")' replaces Bioconductor standard repositories, see
#> 'help("repositories", package = "BiocManager")' for details.
#> Replacement repositories:
#>     CRAN: https://cran.case.edu/
#> [1] TRUE
BiocManager::version()
#> [1] '3.17'
x <- importBugSigDB(version = 'devel', cache = FALSE)
sigs <- bugsigdbr::getSignatures(x, tax.id.type = 'ncbi', tax.level = 'mixed')
#> Error in vapply(spl, function(s) s[length(s)], character(1)): values must be length 1,
#>  but FUN(X[[1]]) result is length 0
sigs <- bugsigdbr::getSignatures(x, tax.id.type = 'ncbi', tax.level = 'species')
#> Error in vapply(spl, function(s) s[length(s)], character(1)): values must be length 1,
#>  but FUN(X[[1]]) result is length 0
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.0 (2023-04-21)
#>  os       Pop!_OS 22.04 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/New_York
#>  date     2023-05-15
#>  pandoc   2.19.2 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package       * version date (UTC) lib source
#>  BiocFileCache   2.8.0   2023-04-25 [1] Bioconductor
#>  BiocManager     1.30.20 2023-02-24 [1] CRAN (R 4.3.0)
#>  bit             4.0.5   2022-11-15 [1] CRAN (R 4.3.0)
#>  bit64           4.0.5   2020-08-30 [1] CRAN (R 4.3.0)
#>  blob            1.2.4   2023-03-17 [1] CRAN (R 4.3.0)
#>  bugsigdbr     * 1.6.1   2023-04-27 [1] Bioconductor
#>  cachem          1.0.8   2023-05-01 [1] CRAN (R 4.3.0)
#>  cli             3.6.1   2023-03-23 [1] CRAN (R 4.3.0)
#>  crayon          1.5.2   2022-09-29 [1] CRAN (R 4.3.0)
#>  curl            5.0.0   2023-01-12 [1] CRAN (R 4.3.0)
#>  DBI             1.1.3   2022-06-18 [1] CRAN (R 4.3.0)
#>  dbplyr          2.3.2   2023-03-21 [1] CRAN (R 4.3.0)
#>  digest          0.6.31  2022-12-11 [1] CRAN (R 4.3.0)
#>  dplyr           1.1.2   2023-04-20 [1] CRAN (R 4.3.0)
#>  evaluate        0.21    2023-05-05 [1] CRAN (R 4.3.0)
#>  fansi           1.0.4   2023-01-22 [1] CRAN (R 4.3.0)
#>  fastmap         1.1.1   2023-02-24 [1] CRAN (R 4.3.0)
#>  filelock        1.0.2   2018-10-05 [1] CRAN (R 4.3.0)
#>  fs              1.6.2   2023-04-25 [1] CRAN (R 4.3.0)
#>  generics        0.1.3   2022-07-05 [1] CRAN (R 4.3.0)
#>  glue            1.6.2   2022-02-24 [1] CRAN (R 4.3.0)
#>  htmltools       0.5.5   2023-03-23 [1] CRAN (R 4.3.0)
#>  httr            1.4.6   2023-05-08 [1] CRAN (R 4.3.0)
#>  knitr           1.42    2023-01-25 [1] CRAN (R 4.3.0)
#>  lifecycle       1.0.3   2022-10-07 [1] CRAN (R 4.3.0)
#>  magrittr        2.0.3   2022-03-30 [1] CRAN (R 4.3.0)
#>  memoise         2.0.1   2021-11-26 [1] CRAN (R 4.3.0)
#>  pillar          1.9.0   2023-03-22 [1] CRAN (R 4.3.0)
#>  pkgconfig       2.0.3   2019-09-22 [1] CRAN (R 4.3.0)
#>  purrr           1.0.1   2023-01-10 [1] CRAN (R 4.3.0)
#>  R6              2.5.1   2021-08-19 [1] CRAN (R 4.3.0)
#>  reprex          2.0.2   2022-08-17 [1] CRAN (R 4.3.0)
#>  rlang           1.1.1   2023-04-28 [1] CRAN (R 4.3.0)
#>  rmarkdown       2.21    2023-03-26 [1] CRAN (R 4.3.0)
#>  RSQLite         2.3.1   2023-04-03 [1] CRAN (R 4.3.0)
#>  rstudioapi      0.14    2022-08-22 [1] CRAN (R 4.3.0)
#>  sessioninfo     1.2.2   2021-12-06 [1] CRAN (R 4.3.0)
#>  tibble          3.2.1   2023-03-20 [1] CRAN (R 4.3.0)
#>  tidyselect      1.2.0   2022-10-10 [1] CRAN (R 4.3.0)
#>  tzdb            0.4.0   2023-05-12 [1] CRAN (R 4.3.0)
#>  utf8            1.2.3   2023-01-31 [1] CRAN (R 4.3.0)
#>  vctrs           0.6.2   2023-04-19 [1] CRAN (R 4.3.0)
#>  vroom           1.6.3   2023-04-28 [1] CRAN (R 4.3.0)
#>  withr           2.5.0   2022-03-03 [1] CRAN (R 4.3.0)
#>  xfun            0.39    2023-04-20 [1] CRAN (R 4.3.0)
#>  yaml            2.3.7   2023-01-23 [1] CRAN (R 4.3.0)
#> 
#>  [1] /home/samuel/R/x86_64-pc-linux-gnu-library/4.3
#>  [2] /home/samuel/Apps/R-4.3.0/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

<sup>Created on 2023-05-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by sdgamboa** [comment]
I couldn't make it work with the devel version either:

``` r
library(bugsigdbr)
# BiocManager::install('bugsigdbr', force = TRUE)
# BiocManager::install('waldronlab/bugsigdbr', force = TRUE)
# BiocManager::valid()
# BiocManager::version()
x <- importBugSigDB(version = 'devel', cache = FALSE)
sigs <- bugsigdbr::getSignatures(x, tax.id.type = 'ncbi', tax.level = 'mixed')
#> Error in vapply(spl, function(s) s[length(s)], character(1)): values must be length 1,
#>  but FUN(X[[1]]) result is length 0
sigs <- bugsigdbr::getSignatures(x, tax.id.type = 'ncbi', tax.level = 'species')
#> Error in vapply(spl, function(s) s[length(s)], character(1)): values must be length 1,
#>  but FUN(X[[1]]) result is length 0
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.0 (2023-04-21)
#>  os       Pop!_OS 22.04 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/New_York
#>  date     2023-05-15
#>  pandoc   2.19.2 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package       * version date (UTC) lib source
#>  BiocFileCache   2.8.0   2023-04-25 [1] Bioconductor
#>  bit             4.0.5   2022-11-15 [1] CRAN (R 4.3.0)
#>  bit64           4.0.5   2020-08-30 [1] CRAN (R 4.3.0)
#>  blob            1.2.4   2023-03-17 [1] CRAN (R 4.3.0)
#>  bugsigdbr     * 1.7.1   2023-05-15 [1] Github (waldronlab/bugsigdbr@b4bb24a)
#>  cachem          1.0.8   2023-05-01 [1] CRAN (R 4.3.0)
#>  cli             3.6.1   2023-03-23 [1] CRAN (R 4.3.0)
#>  crayon          1.5.2   2022-09-29 [1] CRAN (R 4.3.0)
#>  curl            5.0.0   2023-01-12 [1] CRAN (R 4.3.0)
#>  DBI             1.1.3   2022-06-18 [1] CRAN (R 4.3.0)
#>  dbplyr          2.3.2   2023-03-21 [1] CRAN (R 4.3.0)
#>  digest          0.6.31  2022-12-11 [1] CRAN (R 4.3.0)
#>  dplyr           1.1.2   2023-04-20 [1] CRAN (R 4.3.0)
#>  evaluate        0.21    2023-05-05 [1] CRAN (R 4.3.0)
#>  fansi           1.0.4   2023-01-22 [1] CRAN (R 4.3.0)
#>  fastmap         1.1.1   2023-02-24 [1] CRAN (R 4.3.0)
#>  filelock        1.0.2   2018-10-05 [1] CRAN (R 4.3.0)
#>  fs              1.6.2   2023-04-25 [1] CRAN (R 4.3.0)
#>  generics        0.1.3   2022-07-05 [1] CRAN (R 4.3.0)
#>  glue            1.6.2   2022-02-24 [1] CRAN (R 4.3.0)
#>  htmltools       0.5.5   2023-03-23 [1] CRAN (R 4.3.0)
#>  httr            1.4.6   2023-05-08 [1] CRAN (R 4.3.0)
#>  knitr           1.42    2023-01-25 [1] CRAN (R 4.3.0)
#>  lifecycle       1.0.3   2022-10-07 [1] CRAN (R 4.3.0)
#>  magrittr        2.0.3   2022-03-30 [1] CRAN (R 4.3.0)
#>  memoise         2.0.1   2021-11-26 [1] CRAN (R 4.3.0)
#>  pillar          1.9.0   2023-03-22 [1] CRAN (R 4.3.0)
#>  pkgconfig       2.0.3   2019-09-22 [1] CRAN (R 4.3.0)
#>  purrr           1.0.1   2023-01-10 [1] CRAN (R 4.3.0)
#>  R6              2.5.1   2021-08-19 [1] CRAN (R 4.3.0)
#>  reprex          2.0.2   2022-08-17 [1] CRAN (R 4.3.0)
#>  rlang           1.1.1   2023-04-28 [1] CRAN (R 4.3.0)
#>  rmarkdown       2.21    2023-03-26 [1] CRAN (R 4.3.0)
#>  RSQLite         2.3.1   2023-04-03 [1] CRAN (R 4.3.0)
#>  rstudioapi      0.14    2022-08-22 [1] CRAN (R 4.3.0)
#>  sessioninfo     1.2.2   2021-12-06 [1] CRAN (R 4.3.0)
#>  tibble          3.2.1   2023-03-20 [1] CRAN (R 4.3.0)
#>  tidyselect      1.2.0   2022-10-10 [1] CRAN (R 4.3.0)
#>  tzdb            0.4.0   2023-05-12 [1] CRAN (R 4.3.0)
#>  utf8            1.2.3   2023-01-31 [1] CRAN (R 4.3.0)
#>  vctrs           0.6.2   2023-04-19 [1] CRAN (R 4.3.0)
#>  vroom           1.6.3   2023-04-28 [1] CRAN (R 4.3.0)
#>  withr           2.5.0   2022-03-03 [1] CRAN (R 4.3.0)
#>  xfun            0.39    2023-04-20 [1] CRAN (R 4.3.0)
#>  yaml            2.3.7   2023-01-23 [1] CRAN (R 4.3.0)
#> 
#>  [1] /home/samuel/R/x86_64-pc-linux-gnu-library/4.3
#>  [2] /home/samuel/Apps/R-4.3.0/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

<sup>Created on 2023-05-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by sdgamboa** [comment]
It seems that this only happens when importing the 'devel' exports.

``` r
library(bugsigdbr)
# BiocManager::install('bugsigdbr', force = TRUE)
# BiocManager::install('waldronlab/bugsigdbr', force = TRUE)
# BiocManager::valid()
# BiocManager::version()
x <- importBugSigDB(cache = FALSE)
sigs <- bugsigdbr::getSignatures(x, tax.id.type = 'ncbi', tax.level = 'mixed')
sigs <- bugsigdbr::getSignatures(x, tax.id.type = 'ncbi', tax.level = 'species')
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.0 (2023-04-21)
#>  os       Pop!_OS 22.04 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/New_York
#>  date     2023-05-15
#>  pandoc   2.19.2 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package       * version date (UTC) lib source
#>  BiocFileCache   2.8.0   2023-04-25 [1] Bioconductor
#>  bit             4.0.5   2022-11-15 [1] CRAN (R 4.3.0)
#>  bit64           4.0.5   2020-08-30 [1] CRAN (R 4.3.0)
#>  blob            1.2.4   2023-03-17 [1] CRAN (R 4.3.0)
#>  bugsigdbr     * 1.7.1   2023-05-15 [1] Github (waldronlab/bugsigdbr@b4bb24a)
#>  cachem          1.0.8   2023-05-01 [1] CRAN (R 4.3.0)
#>  cli             3.6.1   2023-03-23 [1] CRAN (R 4.3.0)
#>  crayon          1.5.2   2022-09-29 [1] CRAN (R 4.3.0)
#>  curl            5.0.0   2023-01-12 [1] CRAN (R 4.3.0)
#>  DBI             1.1.3   2022-06-18 [1] CRAN (R 4.3.0)
#>  dbplyr          2.3.2   2023-03-21 [1] CRAN (R 4.3.0)
#>  digest          0.6.31  2022-12-11 [1] CRAN (R 4.3.0)
#>  dplyr           1.1.2   2023-04-20 [1] CRAN (R 4.3.0)
#>  evaluate        0.21    2023-05-05 [1] CRAN (R 4.3.0)
#>  fansi           1.0.4   2023-01-22 [1] CRAN (R 4.3.0)
#>  fastmap         1.1.1   2023-02-24 [1] CRAN (R 4.3.0)
#>  filelock        1.0.2   2018-10-05 [1] CRAN (R 4.3.0)
#>  fs              1.6.2   2023-04-25 [1] CRAN (R 4.3.0)
#>  generics        0.1.3   2022-07-05 [1] CRAN (R 4.3.0)
#>  glue            1.6.2   2022-02-24 [1] CRAN (R 4.3.0)
#>  htmltools       0.5.5   2023-03-23 [1] CRAN (R 4.3.0)
#>  httr            1.4.6   2023-05-08 [1] CRAN (R 4.3.0)
#>  knitr           1.42    2023-01-25 [1] CRAN (R 4.3.0)
#>  lifecycle       1.0.3   2022-10-07 [1] CRAN (R 4.3.0)
#>  magrittr        2.0.3   2022-03-30 [1] CRAN (R 4.3.0)
#>  memoise         2.0.1   2021-11-26 [1] CRAN (R 4.3.0)
#>  pillar          1.9.0   2023-03-22 [1] CRAN (R 4.3.0)
#>  pkgconfig       2.0.3   2019-09-22 [1] CRAN (R 4.3.0)
#>  purrr           1.0.1   2023-01-10 [1] CRAN (R 4.3.0)
#>  R6              2.5.1   2021-08-19 [1] CRAN (R 4.3.0)
#>  reprex          2.0.2   2022-08-17 [1] CRAN (R 4.3.0)
#>  rlang           1.1.1   2023-04-28 [1] CRAN (R 4.3.0)
#>  rmarkdown       2.21    2023-03-26 [1] CRAN (R 4.3.0)
#>  RSQLite         2.3.1   2023-04-03 [1] CRAN (R 4.3.0)
#>  rstudioapi      0.14    2022-08-22 [1] CRAN (R 4.3.0)
#>  sessioninfo     1.2.2   2021-12-06 [1] CRAN (R 4.3.0)
#>  tibble          3.2.1   2023-03-20 [1] CRAN (R 4.3.0)
#>  tidyselect      1.2.0   2022-10-10 [1] CRAN (R 4.3.0)
#>  tzdb            0.4.0   2023-05-12 [1] CRAN (R 4.3.0)
#>  utf8            1.2.3   2023-01-31 [1] CRAN (R 4.3.0)
#>  vctrs           0.6.2   2023-04-19 [1] CRAN (R 4.3.0)
#>  vroom           1.6.3   2023-04-28 [1] CRAN (R 4.3.0)
#>  withr           2.5.0   2022-03-03 [1] CRAN (R 4.3.0)
#>  xfun            0.39    2023-04-20 [1] CRAN (R 4.3.0)
#>  yaml            2.3.7   2023-01-23 [1] CRAN (R 4.3.0)
#> 
#>  [1] /home/samuel/R/x86_64-pc-linux-gnu-library/4.3
#>  [2] /home/samuel/Apps/R-4.3.0/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

<sup>Created on 2023-05-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by cmirzayi** [comment]
The error is due to a bug in Study 722, Experiment 1, Signatures 1 and 2 where a taxon is somehow curated as "". `getSignatures()` does not have any checking for empty taxa. It only shows up in the devel version because 722 was just curated a few weeks ago.

This took me 3 hours to find why but I found it!

I'll have a workaround in a moment but basically you just need to filter out this study.

**Comment by cmirzayi** [comment]
Try this:
```library(bugsigdbr)
x <- importBugSigDB(version = 'devel', cache = FALSE)
y <- x[x$Study!="Study 722",]
sigs <- bugsigdbr::getSignatures(y, tax.id.type = 'ncbi', tax.level = 'mixed')

**Comment by sdgamboa** [comment]
Thanks, @cmirzayi!

**Comment by lgeistlinger** [comment]
Thanks @sdgamboa for reporting this and @cmirzayi for the workaround.

This should be fixed in bugsigdbr v1.7.2 (Bioc devel) and bugsigdbr v1.6.2 (Bioc release).

Try:

```
> library(bugsigdbr)
> x <- importBugSigDB(version = 'devel', cache = FALSE)
> sigs <- bugsigdbr::getSignatures(x, tax.id.type = 'ncbi', tax.level = 'mixed')
```

Please confirm that this works and close the issue if this resolves the problem. Thanks!

**Comment by sdgamboa** [comment]
Devel bugsigdbr worked for me. Thanks, @lgeistlinger! Closing this issue.

``` r
library(bugsigdbr)
x <- importBugSigDB(version = 'devel', cache = FALSE)
sigs <- bugsigdbr::getSignatures(x, tax.id.type = 'ncbi', tax.level = 'mixed')
head(lapply(sigs, head))
#> $`bsdb:1/1/1_colorectal-adenoma:conventional-adenoma-cases_vs_controls_UP`
#> [1] "91061"  "1236"   "1654"   "1716"   "1301"   "162289"
#> 
#> $`bsdb:1/1/2_colorectal-adenoma:conventional-adenoma-cases_vs_controls_DOWN`
#> [1] "100883" "1117"  
#> 
#> $`bsdb:1/2/1_Hyperplastic-Polyp:hyperplastic-polyp-cases_vs_controls_UP`
#> [1] "207244" "57037" 
#> 
#> $`bsdb:1/2/2_Hyperplastic-Polyp:hyperplastic-polyp-cases_vs_controls_DOWN`
#> [1] "1236"   "100883" "91347" 
#> 
#> $`bsdb:1/3/1_Colon-Sessile-Serrated-Adenoma/Polyp:Sessile-serrated-adenoma-cases_vs_controls_DOWN`
#> [1] "526524" "1236"  
#> 
#> $`bsdb:1/4/1_colorectal-adenoma:Proximal-conventional-adenoma-cases_vs_controls_UP`
#> [1] "2037" "2049" "1654" "1653" "1716" "1300"
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.0 (2023-04-21)
#>  os       Pop!_OS 22.04 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/New_York
#>  date     2023-05-20
#>  pandoc   2.19.2 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package       * version date (UTC) lib source
#>  BiocFileCache   2.8.0   2023-04-25 [1] Bioconductor
#>  bit             4.0.5   2022-11-15 [1] CRAN (R 4.3.0)
#>  bit64           4.0.5   2020-08-30 [1] CRAN (R 4.3.0)
#>  blob            1.2.4   2023-03-17 [1] CRAN (R 4.3.0)
#>  bugsigdbr     * 1.7.2   2023-05-20 [1] Github (waldronlab/bugsigdbr@d833c70)
#>  cachem          1.0.8   2023-05-01 [1] CRAN (R 4.3.0)
#>  cli             3.6.1   2023-03-23 [1] CRAN (R 4.3.0)
#>  crayon          1.5.2   2022-09-29 [1] CRAN (R 4.3.0)
#>  curl            5.0.0   2023-01-12 [1] CRAN (R 4.3.0)
#>  DBI             1.1.3   2022-06-18 [1] CRAN (R 4.3.0)
#>  dbplyr          2.3.2   2023-03-21 [1] CRAN (R 4.3.0)
#>  digest          0.6.31  2022-12-11 [1] CRAN (R 4.3.0)
#>  dplyr           1.1.2   2023-04-20 [1] CRAN (R 4.3.0)
#>  evaluate        0.21    2023-05-05 [1] CRAN (R 4.3.0)
#>  fansi           1.0.4   2023-01-22 [1] CRAN (R 4.3.0)
#>  fastmap         1.1.1   2023-02-24 [1] CRAN (R 4.3.0)
#>  filelock        1.0.2   2018-10-05 [1] CRAN (R 4.3.0)
#>  fs              1.6.2   2023-04-25 [1] CRAN (R 4.3.0)
#>  generics        0.1.3   2022-07-05 [1] CRAN (R 4.3.0)
#>  glue            1.6.2   2022-02-24 [1] CRAN (R 4.3.0)
#>  htmltools       0.5.5   2023-03-23 [1] CRAN (R 4.3.0)
#>  httr            1.4.6   2023-05-08 [1] CRAN (R 4.3.0)
#>  knitr           1.42    2023-01-25 [1] CRAN (R 4.3.0)
#>  lifecycle       1.0.3   2022-10-07 [1] CRAN (R 4.3.0)
#>  magrittr        2.0.3   2022-03-30 [1] CRAN (R 4.3.0)
#>  memoise         2.0.1   2021-11-26 [1] CRAN (R 4.3.0)
#>  pillar          1.9.0   2023-03-22 [1] CRAN (R 4.3.0)
#>  pkgconfig       2.0.3   2019-09-22 [1] CRAN (R 4.3.0)
#>  purrr           1.0.1   2023-01-10 [1] CRAN (R 4.3.0)
#>  R6              2.5.1   2021-08-19 [1] CRAN (R 4.3.0)
#>  reprex          2.0.2   2022-08-17 [1] CRAN (R 4.3.0)
#>  rlang           1.1.1   2023-04-28 [1] CRAN (R 4.3.0)
#>  rmarkdown       2.21    2023-03-26 [1] CRAN (R 4.3.0)
#>  RSQLite         2.3.1   2023-04-03 [1] CRAN (R 4.3.0)
#>  rstudioapi      0.14    2022-08-22 [1] CRAN (R 4.3.0)
#>  sessioninfo     1.2.2   2021-12-06 [1] CRAN (R 4.3.0)
#>  tibble          3.2.1   2023-03-20 [1] CRAN (R 4.3.0)
#>  tidyselect      1.2.0   2022-10-10 [1] CRAN (R 4.3.0)
#>  tzdb            0.4.0   2023-05-12 [1] CRAN (R 4.3.0)
#>  utf8            1.2.3   2023-01-31 [1] CRAN (R 4.3.0)
#>  vctrs           0.6.2   2023-04-19 [1] CRAN (R 4.3.0)
#>  vroom           1.6.3   2023-04-28 [1] CRAN (R 4.3.0)
#>  withr           2.5.0   2022-03-03 [1] CRAN (R 4.3.0)
#>  xfun            0.39    2023-04-20 [1] CRAN (R 4.3.0)
#>  yaml            2.3.7   2023-01-23 [1] CRAN (R 4.3.0)
#> 
#>  [1] /home/samuel/R/x86_64-pc-linux-gnu-library/4.3
#>  [2] /home/samuel/Apps/R-4.3.0/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

<sup>Created on 2023-05-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
